spyder/app/mainwindow.py

```diff
@@ -385,15 +370,28 @@ def switch_to_plugin(self, plugin, force_focus=None):
         last_plugin = self.last_plugin
         try:
             # New API
-            if (last_plugin is not None
-                    and last_plugin.get_widget().is_maximized
-                    and last_plugin is not plugin):
-                self.layouts.maximize_dockwidget()
+            if (
+                last_plugin is not None
+                and last_plugin.get_widget().get_maximized_state()
+                and last_plugin is not plugin
+            ):
+                maximize_action = self.layouts.maximize_action
+                if maximize_action.isChecked():
+                    maximize_action.setChecked(False)
+                else:
+                    maximize_action.setChecked(True)
         except AttributeError:
             # Old API
-            if (last_plugin is not None and self.last_plugin._ismaximized
-                    and last_plugin is not plugin):
-                self.layouts.maximize_dockwidget()
+            if (
+                last_plugin is not None
+                and self.last_plugin._ismaximized
+                and last_plugin is not plugin
+            ):
+                maximize_action = self.layouts.maximize_action
+                if maximize_action.isChecked():
+                    maximize_action.setChecked(False)
+                else:
+                    maximize_action.setChecked(True)
```

Comment:
Should all this logic be moved to a new method in the `Layout` plugin? A method with a signature like `switch_to_plugin(self, last_plugin, plugin)` ? If that is done, maybe the method could be reused for the `_switch_to_plugin` method at `spyder.plugins.base`?

Reply:
I this it's a good idea to move this method to the Layout plugin because these checks use actions and methods present on it.

However, I don't agree with using it in `spyder.plugins.base`. I mean, I think there's little gain in improving that code given that it's going to be removed soon, once the editor is migrated to the new API.